cmd/up/up.go

```diff
@@ -947,7 +947,9 @@ func (up *upContext) shutdownHybridMode() {
 
 	terminateChildProcess(up.hybridCommand.Process.Pid, pList)
 
-	if err := terminateProcess(up.hybridCommand.Process.Pid); err != nil {
+	p := newOktetoProcess(up.hybridCommand.Process.Pid)
+
+	if err := terminateProcess(p); err != nil {
```

Comment:
nit: how about extract `p.IsProcessSessionLeader()` here and create two different termination methods?: `terminateProcessByKill` vs `terminateProcessBySignal` and use it inside `terminateChildProcess` or something like that? in order to not check recursively if is sessionLeader for child processes, it is always false, right? that's my main concern

Reply:
@AdrianPedriza we should check each process during the recursion because you might have a child process that is also session leader. The simplest case is when you run `bash` inside `bash` you get two processes that are session leaders.

Example (on MacOS from `zsh` I run `bash,` and from it I run `bash` again)

```
❯ bash
bash-5.2$ echo $$
24513
bash-5.2$ bash
bash-5.2$ echo $$
25704
```

Process tree:

```
-+= 24513 andrea bash
 \--= 25704 andrea bash
```

Process details:

```
❯ ps -j 24513
USER     PID  PPID  PGID   SESS JOBC STAT   TT       TIME COMMAND
andrea 24513 24102 24513      0    1 S+   s000    0:00.01 bash

❯ ps -j 25704
USER     PID  PPID  PGID   SESS JOBC STAT   TT       TIME COMMAND
andrea 25704 24513 25704      0    1 S+   s000    0:00.01 bash
```

In both cases PGID === PID, which makes them process leaders, in fact if I run: `﻿kill -SIGTERM 24513` or `﻿kill -SIGTERM 25704` nothing happens